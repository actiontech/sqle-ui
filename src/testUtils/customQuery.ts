import * as domTestingLib from '@testing-library/dom';
import { fireEvent, screen } from '@testing-library/react';

export const getBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  if (temp.length > 1) {
    throw domTestingLib.queryHelpers.getElementError(
      `Found multiple elements with the selector=${selector}`,
      document.body
    );
  }
  return temp[0];
};

export const getAllBySelector = (selector: string, baseElement?: Element) => {
  let temp;
  if (baseElement !== undefined) {
    temp = baseElement.querySelectorAll(selector);
  } else {
    temp = document.querySelectorAll(selector);
  }
  if (temp.length === 0) {
    throw domTestingLib.queryHelpers.getElementError(
      `Unable to find an element by: selector=${selector}`,
      document.body
    );
  }
  return temp;
};

export const getSelectContentByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item-content',
    screen.getByText(label).parentNode?.parentNode as HTMLDivElement
  );
};

export const getSelectValueByFormLabel = (label: string) => {
  return getBySelector(
    '.ant-select-selection-item',
    screen.getByLabelText(label).parentNode?.parentNode as HTMLDivElement
  );
};

/**
 * 模拟选择 Select 组件中下拉框中的某项数据
 * @param 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionText 需要选中的下拉项
 * @param index 通过文本获取所有的节点会有两个(具体信息可以在网页上查看Select 组件中下拉的节点信息), 正常来说其中第二个是需要来需要用户点击的,
 * 所以这里的 index 默认值为 1. 并且这个节点拥有类名 ant-select-item-option-content, 所以在点击前需要使用 expect 来判断是否为正确的节点.
 */
export const selectOptionByIndex = (
  label: string,
  optionText: string,
  index = 1
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = screen.getAllByText(optionText);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }
  const option = options[realIndex];
  expect(option).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

/**
 * 具体功能同 selectOptionByIndex, 不过该函数主要针对下拉框选项为一个节点的情况
 * @param label 表单中 Select 组件外部 Form.Item 的label, 注意该 Form.Item 必须有 name
 * @param optionCls 下拉项节点最外层的 class
 * @param index  这里不同于 selectOptionByIndex 的 index, 其代表的就是真实需要选择的下拉项的序号, 从 0 开始.
 */
export const selectCustomOptionByClassName = (
  label: string,
  optionCls: string,
  index: number
) => {
  fireEvent.mouseDown(screen.getByLabelText(label));
  const options = getAllBySelector(`.${optionCls}`);
  let realIndex = index;
  if (index < 0) {
    realIndex = options.length + index;
  }

  const option = options[realIndex];
  expect(option.parentNode).toHaveClass('ant-select-item-option-content');
  fireEvent.click(option);
};

export const getHrefByText = (name: string) => {
  return screen.getByText(name).getAttribute('href');
};

export const getAllHrefByText = (name: string) => {
  return screen.getAllByText(name).map((v) => v.getAttribute('href'));
};
