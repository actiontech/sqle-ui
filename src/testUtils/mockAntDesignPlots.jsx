import { cloneDeep, isObject } from 'lodash';
const MockPlots = (props) => {
  const cloneProps = cloneDeep(props);

  Object.keys(cloneProps).forEach((key) => {
    if (isObject(cloneProps[key])) {
      cloneProps[key] = JSON.stringify(cloneProps[key]);
    }
  });
  return <div {...cloneProps} />;
};

const Line = MockPlots;
const Pie = MockPlots;
const RadialBar = MockPlots;
const Gauge = MockPlots;
const Column = MockPlots;
const mockRegisterShape = jest.fn();
const G2 = {
  registerShape: jest.fn(),
};

export { Line, Pie, RadialBar, Gauge, Column, G2, mockRegisterShape };

// eslint-disable-next-line import/no-anonymous-default-export
export default {
  Line,
  Pie,
  RadialBar,
  Gauge,
  Column,
  G2,
};
