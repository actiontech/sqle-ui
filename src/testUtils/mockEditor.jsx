const mockEditor = (props) => {
  // https://github.com/react-monaco-editor/react-monaco-editor/issues/176
  const { editorDidMount, ...otherProps } = props;
  return <input {...otherProps} />;
};

export default mockEditor;
