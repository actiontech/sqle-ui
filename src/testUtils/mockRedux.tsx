import { Dictionary } from '../types/common.type';
import * as redux from 'react-redux';
import locale from '../store/locale';
import user from '../store/user';
import userManage from '../store/userManage';
import whitelist from '../store/whitelist';
import nav from '../store/nav';
import { combineReducers, createStore } from 'redux';
import React from 'react';
import { Provider } from 'react-redux';

const reducers = combineReducers({
  locale,
  user,
  userManage,
  whitelist,
  nav,
});

export const storeFactory = (initStore: Dictionary = {}) => {
  return createStore(reducers, initStore);
};

export const mockUseSelector = (store?: any) => {
  const spy = jest.spyOn(redux, 'useSelector');
  spy.mockImplementation((getFn: Function) => {
    return getFn(store);
  });
  return spy;
};

export const mockUseDispatch = () => {
  const scopeDispatch = jest.fn();
  const spy = jest.spyOn(redux, 'useDispatch');
  spy.mockImplementation(() => scopeDispatch);
  return { spy, scopeDispatch };
};

export const mockUseLocation = () => {}


export const CustomProvider: React.FC<{ initStore: Dictionary }> = (props) => {
  return (
    <Provider store={storeFactory(props.initStore)}>{props.children}</Provider>
  );
};
