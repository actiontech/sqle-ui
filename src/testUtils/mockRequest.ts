import { AxiosResponse } from 'axios';
import instance from '../api/instance';
import role from '../api/role';
import ruleTemplate from '../api/rule_template';
import user from '../api/user';
import workflow from '../api/workflow';

export const successData = (data: any, otherData?: any) => {
  return {
    code: 0,
    message: '',
    data,
    ...otherData,
  };
};

export const failedData = (data?: any) => {
  return {
    code: 100,
    message: 'error',
    data,
  };
};

export const resolveImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.resolve({
    status,
    headers,
    config,
    statusText,
    data: successData(data),
  });
};

export const rejectImmediately = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return Promise.reject({
    status,
    headers,
    config,
    statusText,
    data: failedData(data),
  });
};

export const resolveThreeSecond = (
  data: any,
  {
    status = 200,
    headers = {},
    config = {},
    statusText = '',
    otherData = {},
  } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: successData(data, otherData),
      });
    }, 3000);
  });
};

export const resolveErrorThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((res) => {
    setTimeout(() => {
      res({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const rejectThreeSecond = (
  data: any,
  { status = 200, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise<AxiosResponse<any>>((_, rej) => {
    setTimeout(() => {
      rej({
        status,
        headers,
        config,
        statusText,
        data: failedData(data),
      });
    }, 3000);
  });
};

export const throwErrorThreeSecond = (
  error = 'error',
  { status = 500, headers = {}, config = {}, statusText = '' } = {}
) => {
  return new Promise((res) => {
    throw new Error(error);
  });
};

export const mockUseInstance = () => {
  const spy = jest.spyOn(instance, 'getInstanceTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ instance_name: 'instance1' }])
  );
  return spy;
};

export const mockUseInstanceSchema = () => {
  const spy = jest.spyOn(instance, 'getInstanceSchemasV1');
  spy.mockImplementation(() =>
    resolveThreeSecond({ schema_name_list: ['schema1'] })
  );
  return spy;
};

export const mockUseRole = () => {
  const spy = jest.spyOn(role, 'getRoleTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ role_name: 'role_name1' }])
  );
  return spy;
};

export const mockUseRuleTemplate = () => {
  const spy = jest.spyOn(ruleTemplate, 'getRuleTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ rule_template_name: 'rule_template_name1' }])
  );
  return spy;
};

export const mockUseUsername = () => {
  const spy = jest.spyOn(user, 'getUserTipListV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ user_name: 'user_name1' }])
  );
  return spy;
};

export const mockUseWorkflowTemplate = () => {
  const spy = jest.spyOn(workflow, 'getWorkflowTemplateTipsV1');
  spy.mockImplementation(() =>
    resolveThreeSecond([{ workflow_template_name: 'workflow-template-name-1' }])
  );
  return spy;
};
