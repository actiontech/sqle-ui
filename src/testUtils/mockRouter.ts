import * as reactRouter from 'react-router';

export const mockUseHistory = () => {
  const spy = jest.spyOn(reactRouter, 'useNavigate');
  return spy;
};
