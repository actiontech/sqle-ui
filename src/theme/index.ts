import { makeStyles } from '@material-ui/styles';
import { Theme } from '../types/theme.type';
import darkTheme from './dark';
import lightTheme from './light';

enum SupportTheme {
  DARK = 'dark',
  LIGHT = 'light',
}

const ThemeData = {
  [SupportTheme.DARK]: darkTheme,
  [SupportTheme.LIGHT]: lightTheme,
};

const useStyles = makeStyles<Theme>((theme) => ({
  headerBg: {
    backgroundColor: theme.header.background,
    color: theme.header.color,
  },
  editor: {
    border: theme.editor.border,
    paddingRight: 1,
  },
}));

export { SupportTheme, ThemeData };
export default useStyles;
