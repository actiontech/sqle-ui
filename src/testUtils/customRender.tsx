import { RenderResult, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import {
  BrowserRouter,
  Router,
  RouterProps,
  MemoryRouter,
  MemoryRouterProps,
  useLocation,
} from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dictionary } from '../types/common.type';
import { storeFactory } from './mockRedux';
import lightTheme from '../theme/light';
import { ThemeProvider, StyledEngineProvider } from '@mui/material/styles';
import { mount, shallow } from 'enzyme';

type RenderParams = Parameters<typeof render>;
type MountParams = Parameters<typeof mount>;
type ShallowParams = Parameters<typeof shallow>;

export const renderWithRouter = (...[ui, option]: [...RenderParams]) => {
  return render(<BrowserRouter>{ui}</BrowserRouter>, option);
};

export const renderWithRouterAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <BrowserRouter>
      <Provider store={storeFactory(initStore)}>{ui}</Provider>
    </BrowserRouter>,
    option
  );
};

export const renderWithRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>{ui}</Provider>,
    option
  );
};

export const renderHooksWithRedux = <TProps, TResult>(
  hooks: (props: TProps) => TResult,
  storeState: Dictionary = {}
) => {
  return renderHook(hooks, {
    wrapper: Provider,
    initialProps: {
      store: storeFactory(storeState),
    },
  } as any);
};

export const renderWithServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(<Router {...props}>{ui}</Router>, option);
};

export const renderWithMemoryRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(<MemoryRouter {...props}>{ui}</MemoryRouter>, option);
};

export const renderWithTheme = (...[ui, option]: [...RenderParams]) => {
  return render(
    <StyledEngineProvider injectFirst>
      <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
    </StyledEngineProvider>,
    option
  );
};

export const renderWithThemeAndRedux = (
  ...[ui, option, initStore]: [...RenderParams, Dictionary?]
) => {
  return render(
    <Provider store={storeFactory(initStore)}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Provider>,
    option
  );
};

export const renderWithThemeAndServerRouter = (
  ...[ui, option, props]: [...RenderParams, RouterProps]
) => {
  return render(
    <Router {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </Router>,
    option
  );
};

export const renderWithThemeAndRouter = (
  ...[ui, option, props]: [...RenderParams, MemoryRouterProps?]
) => {
  return render(
    <MemoryRouter {...props}>
      <StyledEngineProvider injectFirst>
        <ThemeProvider theme={lightTheme}>{ui}</ThemeProvider>
      </StyledEngineProvider>
    </MemoryRouter>,
    option
  );
};

export const mountWithTheme = (...[ui, option]: [...MountParams]) => {
  return mount(ui, {
    ...option,
    wrappingComponent: ThemeProvider,
    wrappingComponentProps: {
      theme: lightTheme,
    },
  });
};

export const shallowWithRouter = (...[ui, option]: [...ShallowParams]) => {
  return shallow(ui, {
    ...option,
    wrappingComponent: MemoryRouter,
  });
};

export const renderLocationDisplay = (): [
  () => RenderResult,
  () => JSX.Element
] => {
  const LocationComponent = () => {
    const location = useLocation();

    return <div data-testid="location-display">{location.pathname}</div>;
  };

  return [() => renderWithRouter(<LocationComponent />), LocationComponent];
};
