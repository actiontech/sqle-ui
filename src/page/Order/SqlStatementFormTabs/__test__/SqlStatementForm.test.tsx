/* eslint-disable no-console */
import { act, fireEvent, screen } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import Form, { useForm } from 'antd/lib/form/Form';
import { SqlStatementForm } from '..';
import { getBySelector } from '../../../../testUtils/customQuery';
import { renderWithTheme } from '../../../../testUtils/customRender';

import { SupportTheme } from '../../../../theme';
import { useDispatch, useSelector } from 'react-redux';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe('test Order/SqlStatementForm', () => {
  let tempWarnConsole: typeof console.warn;

  beforeAll(() => {
    tempWarnConsole = console.warn;
    console.warn = (params: any) => {
      if (params === `async-validator:`) {
        return;
      }
      tempWarnConsole(params);
    };
  });

  afterAll(() => {
    console.warn = tempWarnConsole;
  });

  beforeEach(() => {
    (useDispatch as jest.Mock).mockImplementation(() => jest.fn());
    (useSelector as jest.Mock).mockImplementation((e) =>
      e({
        user: { theme: SupportTheme.LIGHT },
      })
    );
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  test('should match snapshot', () => {
    const { result } = renderHook(() => useForm());
    const { container } = renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm form={result.current[0]} />
      </Form>
    );
    expect(container).toMatchSnapshot();
  });

  test('should render corresponding form items with current input type', async () => {
    const { result } = renderHook(() => useForm());

    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm form={result.current[0]} />
      </Form>
    );

    expect(screen.getByText('order.sqlInfo.uploadType')).toBeInTheDocument();
    expect(screen.getByLabelText('order.sqlInfo.manualInput')).toHaveAttribute(
      'checked'
    );
    expect(screen.queryByLabelText('order.sqlInfo.sql')).toHaveValue(
      '/* input your sql */'
    );

    fireEvent.change(screen.getByLabelText('order.sqlInfo.sql'), {
      target: { value: 'select (1)' },
    });
    await act(async () => jest.advanceTimersByTime(0));
    expect(screen.getByLabelText('order.sqlInfo.sql')).toBeInTheDocument();
    expect(
      screen.queryByLabelText('order.sqlInfo.sqlFile')
    ).not.toBeInTheDocument();
    expect(
      screen.queryByLabelText('order.sqlInfo.mybatisFile')
    ).not.toBeInTheDocument();

    fireEvent.click(screen.getByLabelText('order.sqlInfo.uploadFile'));
    expect(
      screen.queryByLabelText('order.sqlInfo.sql')
    ).not.toBeInTheDocument();
    expect(screen.getByLabelText('order.sqlInfo.sqlFile')).toBeInTheDocument();
    expect(
      screen.queryByLabelText('order.sqlInfo.mybatisFile')
    ).not.toBeInTheDocument();

    fireEvent.click(screen.getByLabelText('order.sqlInfo.updateMybatisFile'));
    expect(
      screen.queryByLabelText('order.sqlInfo.sql')
    ).not.toBeInTheDocument();
    expect(
      screen.queryByLabelText('order.sqlInfo.sqlFile')
    ).not.toBeInTheDocument();
    expect(
      screen.getByLabelText('order.sqlInfo.mybatisFile')
    ).toBeInTheDocument();

    fireEvent.click(screen.getByLabelText('order.sqlInfo.manualInput'));
    expect(screen.queryByLabelText('order.sqlInfo.sql')).toHaveValue(
      'select (1)'
    );
  });

  test('should remove form value when isClearFormWhenChangeSqlType is equal true and change current sql input', async () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          isClearFormWhenChangeSqlType={true}
        />
      </Form>
    );
    expect(screen.queryByLabelText('order.sqlInfo.sql')).toHaveValue(
      '/* input your sql */'
    );
    fireEvent.change(screen.getByLabelText('order.sqlInfo.sql'), {
      target: { value: 'select (1)' },
    });
    await act(async () => jest.advanceTimersByTime(0));
    fireEvent.click(screen.getByLabelText('order.sqlInfo.uploadFile'));
    fireEvent.click(screen.getByLabelText('order.sqlInfo.manualInput'));
    expect(screen.queryByLabelText('order.sqlInfo.sql')).toHaveValue(
      '/* input your sql */'
    );
  });

  test('should set upload file list to empty when click remove file', async () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          isClearFormWhenChangeSqlType={true}
        />
      </Form>
    );

    fireEvent.click(screen.getByText('order.sqlInfo.uploadFile'));
    const sqlFile = new File(
      [new Blob(['select * from table_test'], { type: 'text/plain' })],
      'test.sql'
    );
    fireEvent.change(screen.getByLabelText('order.sqlInfo.sqlFile'), {
      target: { files: [sqlFile] },
    });

    await act(async () => jest.advanceTimersByTime(0));

    expect(screen.getByText('test.sql')).toBeInTheDocument();

    fireEvent.click(getBySelector('button[title="Remove file"]'));
    await act(async () => jest.advanceTimersByTime(0));

    expect(screen.queryByText('test.sql')).not.toBeInTheDocument();
  });

  test('should hide update mybatis file radio when hideUpdateMybatisFile is equal true', () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          hideUpdateMybatisFile={true}
        />
      </Form>
    );
    expect(
      screen.queryByLabelText('order.sqlInfo.updateMybatisFile')
    ).not.toBeInTheDocument();
  });

  test('should set sql default value when fieldName is not undefined', () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          sqlStatement={'select (1)'}
        />
      </Form>
    );
    expect(screen.queryByLabelText('order.sqlInfo.sql')).toHaveValue(
      'select (1)'
    );
  });

  test('should be used default name when fieldName is undefined', () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          sqlStatement={'select (1)'}
        />
      </Form>
    );

    expect(getBySelector('.ant-radio-group')).toHaveAttribute(
      'id',
      '0_sqlInputType'
    );
    expect(screen.getByLabelText('order.sqlInfo.sql')).toHaveAttribute(
      'id',
      '0_sql'
    );

    fireEvent.click(screen.getByLabelText('order.sqlInfo.uploadFile'));
    expect(screen.getByLabelText('order.sqlInfo.sqlFile')).toHaveAttribute(
      'id',
      '0_sqlFile'
    );

    fireEvent.click(screen.getByLabelText('order.sqlInfo.updateMybatisFile'));
    expect(screen.getByLabelText('order.sqlInfo.mybatisFile')).toHaveAttribute(
      'id',
      '0_mybatisFile'
    );
  });

  test('should set form item name with fieldName', () => {
    const { result } = renderHook(() => useForm());
    renderWithTheme(
      <Form form={result.current[0]}>
        <SqlStatementForm
          form={result.current[0]}
          sqlStatement={'select (1)'}
          fieldName={'1'}
        />
      </Form>
    );

    expect(getBySelector('.ant-radio-group')).toHaveAttribute(
      'id',
      '1_sqlInputType'
    );
    expect(screen.getByLabelText('order.sqlInfo.sql')).toHaveAttribute(
      'id',
      '1_sql'
    );

    fireEvent.click(screen.getByLabelText('order.sqlInfo.uploadFile'));
    expect(screen.getByLabelText('order.sqlInfo.sqlFile')).toHaveAttribute(
      'id',
      '1_sqlFile'
    );

    fireEvent.click(screen.getByLabelText('order.sqlInfo.updateMybatisFile'));
    expect(screen.getByLabelText('order.sqlInfo.mybatisFile')).toHaveAttribute(
      'id',
      '1_mybatisFile'
    );
  });
});
