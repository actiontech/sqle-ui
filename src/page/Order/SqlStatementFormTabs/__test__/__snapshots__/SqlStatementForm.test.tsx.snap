// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Order/SqlStatementForm should match snapshot 1`] = `
<div>
  <form
    class="ant-form ant-form-horizontal"
  >
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class=""
            for="0_sqlInputType"
            title="order.sqlInfo.uploadType"
          >
            order.sqlInfo.uploadType
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <div
                class="ant-radio-group ant-radio-group-outline"
                id="0_sqlInputType"
              >
                <label
                  class="ant-radio-wrapper ant-radio-wrapper-checked ant-radio-wrapper-in-form-item"
                >
                  <span
                    class="ant-radio ant-radio-checked"
                  >
                    <input
                      checked=""
                      class="ant-radio-input"
                      type="radio"
                      value="0"
                    />
                    <span
                      class="ant-radio-inner"
                    />
                  </span>
                  <span>
                    order.sqlInfo.manualInput
                  </span>
                </label>
                <label
                  class="ant-radio-wrapper ant-radio-wrapper-in-form-item"
                >
                  <span
                    class="ant-radio"
                  >
                    <input
                      class="ant-radio-input"
                      type="radio"
                      value="1"
                    />
                    <span
                      class="ant-radio-inner"
                    />
                  </span>
                  <span>
                    order.sqlInfo.uploadFile
                  </span>
                </label>
                <label
                  class="ant-radio-wrapper ant-radio-wrapper-in-form-item"
                >
                  <span
                    class="ant-radio"
                  >
                    <input
                      class="ant-radio-input"
                      type="radio"
                      value="2"
                    />
                    <span
                      class="ant-radio-inner"
                    />
                  </span>
                  <span>
                    order.sqlInfo.updateMybatisFile
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-form-item"
    >
      <div
        class="ant-row ant-form-item-row"
      >
        <div
          class="ant-col ant-form-item-label"
        >
          <label
            class="ant-form-item-required"
            for="0_sql"
            title="order.sqlInfo.sql"
          >
            order.sqlInfo.sql
          </label>
        </div>
        <div
          class="ant-col ant-form-item-control makeStyles-editor-2 ant-col-xs-24 ant-col-sm-12 ant-col-md-10"
        >
          <div
            class="ant-form-item-control-input"
          >
            <div
              class="ant-form-item-control-input-content"
            >
              <input
                aria-required="true"
                height="500"
                id="0_sql"
                language="sql"
                options="[object Object]"
                theme="vs"
                value="/* input your sql */"
                width="100%"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
